decorator.rb

```diff
@@ -0,0 +1,14 @@
+require_relative 'nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  attr_accessor :nameable
```

Comment:
**Optional**
- Please keep in mind that the `attr_accessor` is an association-generator between two classes to identify the relationship type between them. So, it's always common to place it before the initialization. More logical right?